### PR TITLE
Routing NG: Angular location provider patch

### PR DESCRIPTION
**Pull request by dprokop**
Patches Angulars $location provider for backward compatibility

Supported by tests extracted from https://github.com/angular/angular.js/blob/master/test/ng/locationSpec.js

### CODE REVIEW

File: packages/grafana-runtime/src/services/LocationService.ts [hunk 1]
```diff
@@ -134,15 +135,30 @@ class HistoryWrapper implements LocationService {
...
   return {};
 }
 
-export const locationService: LocationService = new HistoryWrapper();
+export let locationService: LocationService = new HistoryWrapper();
+
+/** @internal
+ * Used for tests only
+ */
+export const setLocationService = (location: LocationService) => {
```
Comment (dprokop) [inline packages/grafana-runtime/src/services/LocationService.ts:156]: @torkelo added this internal setter for test purposes. I want a clean history in every test and this seems to be a reasonable way to do it.

File: public/app/angular/bridgeReactAngularRouting.ts [hunk 2]
```diff
@@ -23,62 +23,6 @@ const tamperAngularLocation = () => {
...
-      locationService.push(newUrl);
-    }
-
-    return locationService.getLocation().pathname;
-  }
-}
-
 // Intercepting $location service with implementation based on history
 const interceptAngularLocation = () => {
   // debugger;
```
Comment (dprokop) [inline public/app/angular/bridgeReactAngularRouting.ts:28]:
```suggestion
```

File: public/app/angular/AngularLocationWrapper.ts [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+  url(newUrl?: any) {
+    navigationLogger('AngularLocationWrapper', false, 'Angular compat layer: url');
+
+    if (newUrl !== undefined) {
+      if (newUrl.startsWith('#')) {
+        locationService.push({ ...locationService.getLocation(), hash: newUrl });
+      } else if (newUrl.startsWith('?')) {
+        locationService.push({ ...locationService.getLocation(), search: newUrl });
+      } else if (newUrl.trim().length === 0) {
+        console.log('pushing emptu');
```
Comment (dprokop) [inline public/app/angular/AngularLocationWrapper.ts:115]:
```suggestion
```

File: packages/grafana-runtime/src/services/LocationService.ts (hunk 1 above)
Comment (torkelo) [reply packages/grafana-runtime/src/services/LocationService.ts:156]: :+1:

**Review by torkelo — APPROVED** [review]
Looks good!

**Pull request merged**